### ISSUE TITLE
examples/external/gui.xtm fails 

### ISSUE THREAD

**Issue by tjltjl** [digego/extempore#241]
Trying to run the gui test fails with a strange error:

....[Lots of compilation]...
```
Compiled:  Widget_left_child >>> [Widget*,Widget*]*
Compiled:  Widget_set_left_child >>> [Widget*,Widget*,Widget*]*
Compiled:  Widget_right_child >>> [Widget*,Widget*]*
Compiled:  Widget_set_right_child >>> [Widget*,Widget*,Widget*]*
Compiled:  Widget_print >>> [void,Widget*]*
Compiled:  Widget_toString >>> [String*,Widget*]*
Compiler Error cannot find closure Widget_print
```

**Comment by tjltjl** [comment]
Replacing

`(bind-poly print Widget_print)`

with

`(bind-func print:[void,Widget*]* (lambda (a) (Widget_print a)))`

gets past that error... is that the correct fix?

**Comment by tjltjl** [comment]
...apparently there is some more bitrot in the file, there is an if with only one branch later on, and the compiler complains about it

**Comment by tjltjl** [comment]
...and later on in the file, when compiling gui_render_subtree, I get something really weird:

Type Error bad type notype  ((Widget_cb##506 widget) widget bounds)

**Comment by benswift** [comment]
Yep, it seems like this example has been borked recently by the changes
to xtlang's polymorphism. I've just pushed up some fixes to get it to
compile (8218d22) but there's still something going wrong with
displaying the widgets - will investigate as soon as I can.

**Comment by tjltjl** [comment]
Interestingly, it seems that simply rerunning the definition of gui_render_subtree (i.e. rerunning that one bind-func) fixes the behaviour at runtime.

Something badly wrong with recursion?

**Comment by benswift** [comment]
Seems to be fixed as at bf8d796
